elliottlib/cli/find_bugs_cli.py

```diff
@@ -220,16 +220,52 @@ def _filter_bugs(bugs):  # returns a list of bugs that should be processed
     # Otherwise we don't need to sweep bugs at all.
     if not (into_default_advisories or default_advisory_type):
         return
-    impetus_bugs = {}  # key is impetus ("rpm", "image", "extras"), value is a set of bug IDs.
-    # @lmeyer: simple and stupid would still be keeping the logic in python, possibly with config flags for branched logic. until that logic becomes too ugly to keep in python, i suppose..
+
+    # key is impetus ("rpm", "image", "extras"), value is a set of bug IDs.
+    impetus_bugs = {
+        "rpm": set(),
+        "image": set(),
+        "extras": set()
+    }
+
+    # @lmeyer: simple and stupid would still be keeping the logic in python,
+    # possibly with config flags for branched logic.
+    # until that logic becomes too ugly to keep in python, i suppose..
     if major_version < 4:  # for 3.x, all bugs should go to the rpm advisory
         impetus_bugs["rpm"] = set(bugs)
     else:  # for 4.x
-        # optional operators bugs should be swept to the "extras" advisory, while other bugs should be swept to "image" advisory.
-        # a way to identify operator-related bugs is by its "Component" value. temporarily hardcode here until we need to move it to ocp-build-data.
+        # sweep rpm cve trackers into "rpm" advisory
+        if mode == 'sweep' and cve_trackers:
+            rpm_bugs = bzutil.get_valid_rpm_cves(bugs)
+            green_prefix("RPM CVEs found: ")
+            click.echo([b.id for b in rpm_bugs])
+
+            # check if corresponding brew builds are attached to advisory
+            attached_builds = errata.get_advisory_nvrs(runtime.group_config.advisories["rpm"])
```

Comment:
Does this mean we need to assume builds are already swept before running a bug sweep?
I get a feeling this will become a gotcha if we adopt this assumption.
There are multiple ways to distinguish if a component is an image or rpm. All brew container images are ending with `-container` suffix.

Reply:
Yes, this is a design choice I wasn't sure of myself, and I would prefer to keep `sweep bugs` and `sweep builds` independent. One alternative I looked at was
- get rpm cves and get the package names
- get unreleased nvrs from brew and search for package names in that (this is the sweep build logic)

But this would be pretty heavy, and if anything changes brew-wise between sweep bugs run and sweep build run, then things would be inconsistent. 

I think we should keep the validation optional, and by default attach bugs to the rpm advisory. If at sweep bugs time, more than 0 builds are attached and if package name is not found - then raise warning. If in this flow we notice that this actually doesn't work then we come back and make it not-optional. 

@sosiouxme thoughts?

Reply:
As mentioned in above points, relying on the naming convention is probably better than coupling this to sweep order.

Re "validation optional" also remember that container image CVEs come in with whiteboard `component:openshift-golang-builder-container` instead of the full set of images actually affected; I think we need to tolerate that.

Reply:
Validation is probably necessary at some stage. If naming convention alone is used, it could be possible for tracker bugs to be added to an advisory, without a corresponding build (this seems to happen a lot with the jenkins rpms, which aren't shipped release). It's painful to rectify this, the security role is need to remove security tracking bugs from advisories.

Reply:
sfowl that is a valid case, may I suggest a new command `remove-orphan-bugs` ? This would be independent of both the commands, and would help in manual workflow as well.

But to your point of having the validation step, rethinking about all possible options, the easiest seems for us to sweep builds and then sweep bugs validating with builds. Maybe a `--validate` flag to force validation, runs without it would not have that.

Reply:
If `remove-orphan-bugs` removes bugs from advisories I think that would require the user to have security permissions in ET, which I don't think ART members have. We want members to be able to prepare correct advisories independently.

If validation is optional then I imagine it's possible for bugs to be attached to the wrong advisory, which then requires non-ART members to get involved, which we want to avoid.

Reply:
Thanks for clarifying that. @sosiouxme I've added the `--check-builds` flag to enforce validation, it will skip attaching bugs if builds are not found. Ready for review

Reply:
Thinking about it, a run without this flag would result in rpm bugs still being attached to image advisory. Either we can skip adding those.. or have the flag always on.?

Reply:
It's interesting that the cost of adding bugs is low but removing is high.. so we really have to get it right the first time or skip to allow for manual addition